### ISSUE TITLE
`!!` should be disallowed

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#29902]
```
        // PROTOTYPE(NullableReferenceTypes): Binder should report an error for `!!`.
        [Fact]
        public void SuppressNullableWarning_Multiple()
        {
            var source =
@"class C
{
    static void F(string? s)
    {
        G(default!!);
        G(s!!);
        G((s!)!);
    }
    static void G(string s)
    {
    }
}";
            var comp = CreateCompilation(
                new[] { source, NonNullTypesTrue, NonNullTypesAttributesDefinition },
                parseOptions: TestOptions.Regular8);
            comp.VerifyDiagnostics(/* ... */);
        }
```

**Comment by jcouv** [comment]
Relates to https://github.com/dotnet/roslyn/issues/25372 (warn for un-necessary `!`)

**Comment by gafter** [comment]
Was there an LDM decision to support this?

**Comment by jcouv** [comment]
I've updated the linked issue (warn for un-necessary `!`) after checking the LDM notes from last week. There was an explicit decision from LDM not to warn for such cases.
Regarding `!!`, there was no explicit decision from LDM yet.

**Comment by gafter** [comment]
I’m pretty sure that the decision not to warn included a decision not to error.

**Comment by jcouv** [comment]
The rationale for not warning (namely so that copy&paste doesn't produce annoying warnings) doesn't apply to '!!' which would be an error in any context.

**Comment by gafter** [comment]
Wouldn’t copy-and-paste (e.g. to inline a variable) now possibly cause an annoying error?

**Comment by jcouv** [comment]
Yes, that could happen.
Then, are you proposing to close this issue as "by design"?

**Comment by gafter** [comment]
Yes, I think it is permitted by design and not harmful.

**Comment by cston** [comment]
`!!` should be allowed. Need tests.
